### DISCUSSION TITLE
Decimal() and division

### DISCUSSION THREAD

**Discussion by masta79** [duckdb/duckdb#8780] [General]
Is there a way to force division to be done as DECIMAL()?

When I'm doing any other operation with DECIMAL numbers, the result will also be a decimal, only division will force the result to be a double, even division between two DECIMALs.

**Comment by Tishj** [comment]
I think that's the difference between `/` and `//`

**Reply by masta79** [reply]
This is what I also assumed, but `//` behaves just like `/`. See the following example.

```sql
D>  17::DECIMAL(18,5)/5.1::DECIMAL(18,5) as slash_division, 17::DECIMAL(18,5)//5.1::DECIMAL(18,5) as double_slash_division, (17::DECIMAL(18,5)//5.1::DECIMAL(18,5))::DECIMAL(18,5
```

|   slash_division<br/>double   | double_slash_division <br/> double| double_slash_division_casted <br/>decimal(18,5) |
|--------------------|-----------------------|------------------------------|
| 3.3333333333333335 | 3.3333333333333335    | 3.33333                      |
